src/Interactive/Features/Interactive/Core/InteractiveHost.cs

```diff
@@ -40,6 +40,8 @@ internal sealed partial class InteractiveHost : MarshalByRefObject
 
         private TextWriter _output;
         private TextWriter _errorOutput;
+        private object _outputGuard;
+        private object _errorOutputGuard;
```

Comment:
`readonly` for both. And do we need two lock objects or could the same object be used for both `_output` and `_errorOutput`?

Reply:
Thanks!
1. Added readonly
2. Discussed with Tomas to use a common lock or separate ones and found that separate ones should be better to avoid two channels to be executed independently.

---
In reply to: [188322002](https://github.com/dotnet/roslyn/pull/26770#discussion_r188322002) [](ancestors = 188322002)